### DISCUSSION TITLE
Context menu MRU

### DISCUSSION THREAD

**Discussion by cheater** [YoshifumiFuyuno/Search-from-Popup-or-ContextMenu#40] [Ideas]
Add 5 most recently used searches to context menu. Make multiple-searches show up in that list as well. make it possible to give the multiple-searches a title for this purpose. Make the MRU show up at the very top.

Make it possible to turn this feature off.

Make it possible to set the amount of MRU (1-5).

Do not show them in a sub-menu - show them at the top level. An option for displaying MRU in a sub menu could be useful for some users.

(MRU = Most Recently Used)

**Comment by YoshifumiFuyuno** [comment]
Due to the limitation of WebExtension, it is impossible to display multiple items at the top level.

**Comment by cheater** [comment]
Sorry, we misunderstood each other. I meant at the top level of the sub-menu that SPC creates.

**Comment by YoshifumiFuyuno** [comment]
If there is only one, you can use `lastUsedEngine`.
cf. [Bookmarklet APIs](https://github.com/YoshifumiFuyuno/Search-from-Popup-or-ContextMenu/wiki/Bookmarklet-APIs)

**Reply by cheater** [reply]
I would like to set the amount between 0 and 5 to see what works best. But I know it'll probably be at least 2. For example this week I've spent several hours searching on two search engines.

**Comment by cheater** [comment]
One thing I'd like to do that I don't think is possible using the Bookmarklet API is that I would like for the icon to be the same as the original menu entry. I'd also like for the name to be as the original menu entry, but also have my chosen accelerator added (eg 1-5 or qwert). And if there already is an accelerator in the original title, then it should be removed. Remove the accelerator using this rule:

1. if it's part of a word, like `&Amazon` or `A&mazon`, then just remove the ampersand `&`
2. if it's next to a single character in parentheses, like `(&X)`, then remove the whole `(&X)` complete with parentheses, and a preceding space. For example `Amazon (&X)` -> `Amazon` (no space at the end), but also `Amazon(&X)` -> `Amazon`
3. if it's next to a single character separated by a space, then remove that. For example `Amazon &X` -> `Amazon` (no space at the end). However, bear in mind you'd still need to do `Amazon&X` -> `AmazonX` because rule 1 applies to that.

This can be done using regex, I can write it for you if you'd like.

Then, once the title is stripped like that, add ` (&q)` at the end for the latest engine, ` (&w)` for the second-latest, and so on, until the 5th.
